src/eth-config-device.cpp

```diff
@@ -0,0 +1,246 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2025 RealSense, Inc. All Rights Reserved.
+
+#include "eth-config-device.h"
+
+#include <rsutils/string/from.h>
+#include <rsutils/type/eth-config.h>
+#include <librealsense2/h/rs_eth_config.h>
+
+namespace librealsense
+{
+    // Command opcodes from rs-dds-config tool
+    constexpr uint32_t GET_ETH_CONFIG = 0xBB;
+    constexpr uint32_t SET_ETH_CONFIG = 0xBA;
+    constexpr uint32_t ETH_CONFIG_DEFAULT = 0;  // For factory reset
+    constexpr uint32_t ETH_CONFIG_CURRENT = 1;  // For current values
+
+    void eth_config_device::init( debug_interface * debug )
+    {
+        if( ! debug )
+            throw std::runtime_error( "eth_config_device::init called with null parameter" );
+
+        _debug = debug;
+        if( supports_ethernet_configuration() )
+        {
+            _supported = true;
+            _curr_config = read_eth_config_from_device();
+        }
+    }
+
+    bool eth_config_device::supports_ethernet_configuration()
+    {
+        try
+        {
+            read_eth_config_from_device(); // Test if we can successfully get current configuration
```

Comment:
do you wanty to check here that the size of the returned data is above 0? (or more)